### PR TITLE
Options must be an object

### PR DESCRIPTION
**Pull request by ptomato**
See tc39/ecma402#480 and tc39/ecma402#493. This seems to have critical mass over on the ECMA-402 side now, so I'm opening this PR again.

### CODE REVIEW

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -1631,9 +1633,14 @@ export const ES = ObjectAssign({}, ES2019, {
     return new TemporalTimeZone(ES.TemporalTimeZoneFromString(fmt.resolvedOptions().timeZone));
   },
   ComparisonResult: (value) => (value < 0 ? -1 : value > 0 ? 1 : value),
+  NormalizeOptionsObject: (options) => {
+    if (options === undefined) return {};
+    if (options !== null && (typeof options === 'object' || typeof options === 'function')) {
+      return options;
+    }
```
Comment (ljharb) [inline polyfill/lib/ecmascript.mjs:1640]:
```suggestion
    if (ES.Type(options) === 'Object') {
      return options;
    }
```

Comment (ptomato) [reply polyfill/lib/ecmascript.mjs:1640]: Thanks for the suggestion, that makes things easier to follow.

File: polyfill/lib/ecmascript.mjs [hunk 2]
```diff
@@ -1581,9 +1584,14 @@ export const ES = ObjectAssign({}, ES2019, {
     return new TemporalTimeZone(ES.TemporalTimeZoneFromString(fmt.resolvedOptions().timeZone));
   },
   ComparisonResult: (value) => (value < 0 ? -1 : value > 0 ? 1 : value),
+  NormalizeOptionsObject: (options) => {
+    if (options === undefined) return {};
+    if (options !== null && (typeof options === 'object' || typeof options === 'function')) {
```
[the remark this answers is not in the document]
Comment (ptomato) [reply polyfill/lib/ecmascript.mjs]: I think #256 sufficiently covers that?

File: spec/abstractops.html [hunk 3]
```diff
@@ -21,17 +21,21 @@ <h1>NormalizeOptionsObject ( _options_ )</h1>
...
     </p>
 
     <emu-alg>
       1. Assert: Type(_options_) is Object.
       1. Let _value_ be ? Get(_options_, _property_).
       1. If _value_ is *undefined*, return _fallback_.
-      1. Let _value_ be ? ToString(_value_).
+      1. Assert: _type_ is *"boolean"* or *"string"*.
+      1. If _type_ is *"boolean"*, then
+        1. Let _value_ be ToBoolean(_value_).
```
[the remark this answers is not in the document]
Comment (ptomato) [reply spec/abstractops.html:36]: Added this to https://github.com/tc39/ecma402/pull/493 as well.

**Comment by ptomato** [comment]
@Ms2ger Weirdly, I didn't get a notification for your review. Most of the comments I had already fixed but you caught some that I didn't catch, so thanks!

**Review by Ms2ger — COMMENTED** [review]
LGTM, but this wants a boatload of tests

**Comment by ptomato** [comment]
Tests added.

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**